pkg/kubecost/service/allocation/allocationsummary.proto

```diff
@@ -0,0 +1,147 @@
+
+option go_package = "github.com/opencost/opencost/pkg/kubecost/service/allocation/allocationsummary";
+
+import "google/protobuf/timestamp.proto";
+import "google/protobuf/duration.proto";
+
+// Allocation Summary Interface exported by the gRPC server.
+service AllocationSummary {
+    // An RPC for allocation.
+    //
+    // Obtains the allocation summary for a given window span.
+    
+    // Obtains the AllocationSummarySets satisfy the given AllocationSummaryRequest. Results are
+    // streamed rather than returned at once.
+    rpc GetAllocationSummary (AllocationSummaryRequest) returns (stream AllocationSummaryResponse) {}
+}
+
+// AllocationSummaryRequest is the request send from client with a specify window
+// either absolute window with datetime or relative window with duration
+message AllocationSummaryRequest {
+    Window window =1;
+    repeated string aggregate =2;
+    bool accumulate=4;
+    FilterSelection filter = 5;
+    SharedSelection shared = 6;
+    IdleSelection idle = 7;
+    bool external =8;
+}
+
+// Window represents the window allocation summary is requested
+message Window{
+    oneof window{
+        AbsoluteWindow absWindow=1;
+        RelativeWindow relWindow=2;
+    }
+}
+
+// Absolute Window represents the absolute window with start and end time
+// for 
+message AbsoluteWindow {
+    google.protobuf.Timestamp start = 1;
+    google.protobuf.Timestamp end=2;
+}
+
+//Relative window represents duration for the requested data
+message RelativeWindow {
+    google.protobuf.Duration timeFrame = 1;
+}
+
+// Filter selection represents the filter on the summary data to be returned
+message FilterSelection{
+    repeated string filterClusters=1;
+    repeated string filterNodes = 2;
+    repeated string filterNamespaces = 3; 
+    repeated string filterControllerKinds = 4;
+    repeated string filterController = 5;
+    repeated string filterPods = 6;
+    repeated string filterAnnotations = 7;
+    repeated string filterLabels = 8;
+    repeated string filterServices = 9;
+}
+
+// Different selection of idle from the User Interface 
+enum IdleSelection {
+    IDLE_COST_HIDE = 0;
+    IDLE_COST_SEPERATE = 1;
+    IDLE_COST_SHARE_BY_CLUSTER = 2;
+    IDLE_COST_SHARE_BY_NODE = 3;
+}
+
+// Different shared selection available for the user on the User Interface
+message SharedSelection {
+    bool shareTenancyCosts = 1;
+    string shareNamespaces= 2;
+    string shareLabels = 3;
+    double shareCost = 4;
+    ShareSplit shareSplit = 5;
+}
+
+// Enum for the shared split which can either be weighted or even
+enum ShareSplit {
+    SHARESPLIT_WEIGHTED = 0;
+    SHARESPLIT_EVEN = 1;
+}
+
+// message for the allocationsummary response can either be partial response with each graphdata
+// or complete allocation response having all the details
+message AllocationSummaryResponse {
+    oneof data{
+        AllocationProgressUpdate status_update = 1;
+        CompleteAllocationResponse full_response = 2;
```

Comment:
Should this be "FinalAllocationResponse" - as it's not the complete picture (since it doesn't have the bargraph data)